packages/std/src/deps.rs

```diff
@@ -50,4 +50,49 @@ impl<'a> DepsMut<'a> {
             querier: self.querier,
         }
     }
+
+    pub fn dup(&'_ mut self) -> DepsMut<'_> {
+        DepsMut {
+            storage: self.storage,
+            api: self.api,
+            querier: self.querier,
+        }
+    }
+}
+
+impl<'a> Deps<'a> {
+    pub fn dup(&'_ self) -> Deps<'_> {
+        Deps {
+            storage: self.storage,
+            api: self.api,
+            querier: self.querier,
+        }
+    }
```

Comment:
Do I understand correctly that the lifetime of the result is <= the lifetime of self? So `Deps::clone` (or even copy) can always be used instead, providing a "better" result because it lives as long as `'a`? Then this is only for consistency with DepsMut, right?

Assuming I understand this correctly, I'd rename to `pub fn branch`, indicating that the result has no life independent of the parent.

Reply:
You do understand correctly.

Reviewing, I see I do not need for Deps as it has Clone, but I need for DepsMut only.

The goal is to be able to call a function that needs DepsMut, without moving it, so we can continue accessing the storage later on.

DepsMut::branch() works for me as naming. I will update the pr